crates/runtime/src/dataconnector/spiceai.rs

```diff
@@ -159,6 +163,24 @@ impl DataConnector for SpiceAI {
         Some(read_write_result)
     }
 
+    fn supports_append_stream(&self) -> bool {
+        true
+    }
+
+    async fn append_stream(&self, table_provider: Arc<dyn TableProvider>) -> Option<ChangesStream> {
+        let federated_table_provider_adaptor = table_provider
+            .as_any()
+            .downcast_ref::<FederatedTableProviderAdaptor>()?;
+        let flight_table = federated_table_provider_adaptor
+            .table_provider
+            .as_ref()
+            .unwrap()
```

Comment:
remove `unwrap()`

Reply:
This code should live in this data connector anyway, so this downcast likely goes away